device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/transport/amqps/AmqpsIotHubConnection.java

```diff
@@ -194,8 +169,20 @@ public void reset(ScheduledExecutorService scheduledExecutorService) throws Tran
      *
      * @throws TransportException If the reactor could not be initialized.
      */
-    public void open(Queue<DeviceClientConfig> deviceClientConfigs) throws TransportException
+    public void open(Queue<DeviceClientConfig> deviceClientConfigs, ScheduledExecutorService scheduledExecutorService) throws TransportException
     {
+        reconnectionScheduled = false;
+        connectionId = UUID.randomUUID().toString();
+
+        this.scheduledExecutorService = scheduledExecutorService;
+
+        this.closeLatch = new CountDownLatch(1);
+        this.openLatch = new CountDownLatch(1);
+        this.savedException = null;
+
+        this.state = IotHubConnectionStatus.DISCONNECTED;
```

Comment:
Shouldn't this method check the state of the connection object before setting it to DISCONNECTED? What if it was already connected here?

Reply:
fixed